### PR TITLE
disallow-space-before-commas

### PR DESCRIPTION
**Pull request by ShashankaNataraj**
Adds a new rule to disallow spaces before commas.

Issue is raised here: https://github.com/jscs-dev/node-jscs/issues/1447

### CODE REVIEW

**Comment by markelog** [comment]
Looks good, opposite rule should be pretty easy then?

**Comment by ShashankaNataraj** [comment]
@markelog yep, oppositte rule should be pretty easy. I will take that up too?

**Comment by markelog** [comment]
> would we prefer to write a seperate rule allow-space-before-commas 

Sounds good! You can add a separate commit to this PR, would you be able to do this today, would like these rules to be in 2.0, which should happen today.

**Comment by ShashankaNataraj** [comment]
@markelog @hzoo here we go :dancer: :+1: Please review and accept if this PR is OK.

File: lib/config/configuration.js [hunk 1]
```diff
@@ -985,6 +985,10 @@ Configuration.prototype.registerDefaultRules = function() {
     this.registerRule(require('../rules/require-matching-function-name'));
 
     this.registerRule(require('../rules/disallow-space-before-semicolon'));
+
+    this.registerRule(require('../rules/disallow-space-before-comma'));
+
```
Comment (markelog) [inline lib/config/configuration.js]: We can remove that newline

**Comment by markelog** [comment]
> Coverage decreased (-1.0%)

It seems we need more tests, you can check it out more thoroughly with `npm run coverage-html && open coverage.html` command

**Comment by ShashankaNataraj** [comment]
@markelog I did `npm run coverage-html` and then opened coverage.html in chrome. The below screenshots show that the coverage is 100% for the files which are included in this PR. Am I missing something?

screenshot from 2015-07-21 07 59 32
screenshot from 2015-07-21 07 59 46

**Comment by mrjoelkemp** [comment]
I restarted the build to see if the coveralls report changes. It's claiming the token assertion lib dropped in coverage though you're not really touching it: https://coveralls.io/builds/3102954/source?filename=lib%2Ftoken-assert.js

1. This PR is missing fix tests.

**Comment by ShashankaNataraj** [comment]
@mrjoelkemp Never used coveralls before, so dont really know how to fix this issue. Can you please elaborate on what needs to be done to fix this? Or, can this PR be accepted because none of the files Im working with on this PR are the causes of the break?

**Comment by mrjoelkemp** [comment]
@ShashankaNataraj Mind rebasing off of master and then force pushing up?

**Comment by ShashankaNataraj** [comment]
@markelog I did, now I see all checks have failed. I also observed that builds on other issues are failing as well.

**Comment by markelog** [comment]
Thank you!

**Comment by ShashankaNataraj** [comment]
@markelog Hi, Was this PR merged? I can see that it is closed now, but does it count as a contribution of mine?

Just confused I guess because this is the first PR of mine on github which has "markelog closed this pull request from a commit 14 days ago" text. Dont know what it means, thus the confusion!

**Comment by markelog** [comment]
> @markelog Hi, Was this PR merged?

Yep and it already used in couple of presets

> I can see that it is closed now, but does it count as a contribution of mine?

It sure does :-).

> "markelog closed this pull request from a commit 14 days ago"

This means i pushed your commit to the repo, since you don't have push privileges, but commit is credited to you - https://github.com/jscs-dev/node-jscs/commit/66f89940c01d86dad0017213888f68f508e769fb

**Comment by ShashankaNataraj** [comment]
@markelog Ah ok. Thanks for the clarification!

**Pull request closed without merging**